count-things/javascript/source-type.js

```diff
@@ -0,0 +1,15 @@
+import { join } from 'path';
+import { existsSync } from 'fs';
+import { readFile } from 'fs/promises';
+
+// not currently in use
+//  package.json is not a reliable way to detect source type
```

Comment:
it might be easier to separate the different source types in a "data cleaning" step, so that we can predict which analysis will run on which type of source, but we'll get into the data cleaning bits later, I imagine.

Reply:
this [series of conditionals and try/catches](https://github.com/learning-as-code/corpus-analysis/blob/680b8760cab262af974fcc0bbd24f74a42883e9a/count-things/javascript/analyze-source.js#L45) is a crude module check, the results report if each file was a script (didn't include `import` or `export`) or a module

Better solutions for this function would be to remove it, but I kept it as a reminder of this strategy and why it didn't work